addons/stock/wizard/stock_return_picking.py

```diff
@@ -87,9 +87,10 @@ def _create_returns(self, cr, uid, ids, context=None):
         returned_lines = 0
 
         # Cancel assignment of existing chained assigned moves
+        return_move_ids = [move.move_id.id for move in data_obj.browse(cr, uid, data['product_return_moves'], context=context)]
         moves_to_unreserve = []
         for move in pick.move_lines:
-            to_check_moves = [move.move_dest_id] if move.move_dest_id.id else []
+            to_check_moves = [move.move_dest_id] if move.move_dest_id.id and move.id in return_move_ids else []
```

Comment:
It should be return_moves = list(set(move.move_id for ....]
and then : for move in return_moves: